### PR TITLE
Rain-13354: use unversioned memmove to aid arm64 compilation

### PR DESCRIPTION
**Pull request by alok-lw**
memmove_2.2.5 was being used till now. The glibc had issues with memcpy where memcpy_2.14 was mandated so the machines having older glibc versions were getting erorrs. But there has been no history of issues with memmove so we should be able to use unversioned
Just a side note, glibc2.14 was release 9 yrs ago and the current version is 2.32. We shouldn't expect issues with machines running pre 2.14.
RAIN-13354

### CODE REVIEW

**Review by belyalov-lw — APPROVED** [review]
I wonder why did we need it before? what is the story?

**Comment by alok-lw** [comment]
> I wonder why did we need it before? what is the story?

Well it predates me :). The info I gathered was from the comments (link) which I removed in this PR.
The memcpy signature was changed in 2.14 and after that glibc mandated the use of memcpy_2.14 so it broke on systems which had pre 2.14 glibc
But in our case we are not using memcpy, we are using memmove and there is no documented problem with memmove.

File: pcap.go [hunk 1]
```diff
@@ -9,17 +9,12 @@ package pcap
...
 #ifdef  __GLIBC__
-// See this for glibc 2.14 hack below
-// https://www.win.tue.nl/~aeb/linux/misc/gcc-semibug.html
-
-void *__memcpy_glibc_2_2_5(void *, const void *, size_t);
-
 // We use memmove rather than memcpy here since it is safer
 // for overlapping memory regions.
-asm(".symver __memcpy_glibc_2_2_5, memmove@GLIBC_2.2.5");
+
```
Comment (rakeshlacework) [inline pcap.go:14]: can pcap.go be made arch specific?

Comment (alok-lw) [reply pcap.go:14]: Duplicating the whole file for memmove seems not ideal.
I could not find a reason why memmove@2.2.5 was being used. I read the link in the code and it points to a problem in memcpy not memmove. Even memcpy problem would happen in machines who have glibc version < 2.14 which was released 9 yrs ago.
Is there a reason we should preserve this versioned symbol?

Comment (alok-lw) [reply pcap.go:14]: glibc 2.2.5 was released in 2002 and we are pinning memmove to that for no apparent reason ? :)

Comment (rakeshlacework) [reply pcap.go:14]: Correctness is more important here. The onus is to ensure memmove of "current" glibc is not going to lead us into data correctness problem both for the memmove and memcpy usage.

Comment (alok-lw) [reply pcap.go:14]: why do you think current memmove will lead us into correctness issue vs 18 yr old one?

Comment (rakeshlacework) [reply pcap.go:14]: Is there no change in memmove/memcpy since GLIBC 2.2.5?

Comment (alok-lw) [reply pcap.go:14]: does it matter? Pinning to an older version is a bigger risk when we don't have a reason to.
I have tested connections with this change and it works fine.

Comment (rakeshlacework) [reply pcap.go:14]: I have moved this conversation to the bug because it belongs there

**Pull request merged**